### DISCUSSION TITLE
Error: '@itwin/web-viewer-react' has no exported member named 'ItwinViewerUi

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#5560] [Q&A]
I am upgrading our code to 3.7.6 (from 3.5.2) and I am getting the following error.   
```
ERROR in src/AppUiConfig.tsx:5:10
TS2724: '"@itwin/web-viewer-react"' has no exported member named 'ItwinViewerUi'. Did you mean 'ItwinViewer'?
```
I was using the ITwinViewUi class to control UI options in my app.  This was being passed as ```defaultUiConfig={appUiConfig}``` argument in ```<Viewer/>``` component.   Can you please guide what I need to do in 3.7.6 to achieve same requirement.

Thanks,

Ovais

```
/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
import { ItwinViewerUi} from "@itwin/web-viewer-react";

export const appUiConfig: ItwinViewerUi = {
  contentManipulationTools: {
    cornerItem: {
      hideDefault: true,
    },
    hideDefaultHorizontalItems: true,
    hideDefaultVerticalItems: true,
    verticalItems: {
      sectionTools: false,
      measureTools: false,
      selectTool: false,
    },
    horizontalItems: {
      clearSelection: true,
      clearHideIsolateEmphasizeElements: true,
      isolateElements: true,
      hideElements: true,
      emphasizeElements: true,
    },
  },
  navigationTools: {
    hideDefaultHorizontalItems: true,
    hideDefaultVerticalItems: true,
    verticalItems: {
      walkView: true,
      cameraView: false,
    },
    horizontalItems: {
      rotateView: true,
      panView: true,
      fitView: true,
      windowArea: true,
      undoView: true,
      redoView: true,
    },
  },
  hideDefaultStatusBar: false,
  hidePropertyGrid: true,
  hideToolSettings: true,
  hideTreeView: true,
};
```

**Comment by aruniverse** [comment]
hi @ovaisa , 

What version of itwin/web-viewer-react are you using? I'd recommend taking a look at the release notes for our last major version which included some breaking changes wrt to the default viewer ui

https://github.com/iTwin/viewer/blob/master/releases/CHANGELOG-3.0.md

**Reply by ovaisa** [reply]
The change log talks about removing properties from ```defaultUiConfig``` but I was not able to find any additional details  

Version info for itwin/web-viewer-react (from respective package.json files)

3.5.2: "@itwin/web-viewer-react": "^2.0.11"
3.7.6: "@itwin/web-viewer-react": "^3.1.5"

**Reply by aruniverse** [reply]
Previously in web-viewer-react@2.x, we used `defaultUiConfig` to control what features are added to the viewer. As mentioned in the changelog, we removed all the features from the viewer component by default, and recommend users add whatever they need via UiItemsProviders.

```jsx
import { MeasureTools,  MeasureToolsUiItemsProvider } from "@itwin/measure-tools-react";
import {  PropertyGridManager,  PropertyGridUiItemsProvider } from "@itwin/property-grid-react";
import {  TreeWidget,  TreeWidgetUiItemsProvider } from "@itwin/tree-widget-react";
import {  Viewer,  ViewerContentToolsProvider,  ViewerNavigationToolsProvider,  ViewerStatusbarItemsProvider } from "@itwin/web-viewer-react";
import React, { useCallback, useMemo, useState } from "react";

export const App: React.FC = () => {
...

  const onIModelAppInit = useCallback(async () => {
    await TreeWidget.initialize();
    await PropertyGridManager.initialize();
    await MeasureTools.startup();
  }, []);

  return (
    <div style={{ height: "100vh" }}>
      <Viewer
        authClient={authClient}
        iTwinId={iTwinId}
        iModelId={iModelId}
		...
        onIModelAppInit={onIModelAppInit}
        uiProviders={[
          new ViewerNavigationToolsProvider(),
          new ViewerContentToolsProvider(),
          new ViewerStatusbarItemsProvider(),
          new TreeWidgetUiItemsProvider(),
          new PropertyGridUiItemsProvider({
            enableCopyingPropertyText: true,
          }),
          new MeasureToolsUiItemsProvider(),
        ]}
      />
    </div>
  );
};
```

Take a look at every providers constructor which will allow you to select features to enable.

**Reply by ovaisa** [reply]
I was using ```defaultUiConfig``` to mostly manage default toolbars. 

Does it mean I have to now load all toolbars myself as UiItemProviders and then change the default items?   I see that now there are new UiItemProviders being passed to ```<Viewer/>``` 

```
    new ViewerNavigationToolsProvider(),
    new ViewerContentToolsProvider(),
    new ViewerStatusbarItemsProvider(),
```

How can I then modify the default toolbars which I was able to do with ```defaultUiConfig```.   I was not able to find documentation to these on https://www.itwinjs.org/reference/.  No documentation for ```<Viewer/>``` either on this site.  Is this documentation at a different place?  Can you please guide.

**Reply by aruniverse** [reply]
We are working on getting more documentation for the Viewer onto the itwinjs.org site.
Since we use Typescript, we have so far been relying on our users using the `*.d.ts` files.
If you look at the constructors for each of those providers, you will see it accepts an object which you can use to enable the feature you'd like to use.

For example:
[ViewerContentToolsProvider](https://github.com/iTwin/viewer/blob/master/packages/modules/viewer-react/src/components/app-ui/providers/ViewerContentToolsProvider.tsx#L17)
<img width="624" alt="image" src="https://github.com/iTwin/itwinjs-core/assets/11051042/dcf290f3-ecac-41e6-9840-e874429fe4c2">
<img width="893" alt="image" src="https://github.com/iTwin/itwinjs-core/assets/11051042/620ee119-ff62-41ca-b32a-408e30de931f">

**Reply by ovaisa** [reply]
Thanks Arun, I am able to start getting things back.  I see that a new 'Tool Settings' toolbar is added to the UI, I couldn't figure out how to disable this.  I removed all UiItemProviders but this still shows up.  Can you please guide how I can remove this.

![image](https://github.com/iTwin/itwinjs-core/assets/83007711/adfad468-238a-4830-8dde-c38aa607b7b0)

Thanks,

Ovais

**Reply by NancyMcCallB** [reply]
Hi, @ovaisa ,

The Tool Settings is not new to V3, but you may have been running in the obsolete UI 1.0 mode up to this point. You can't disable the Tool Settings bar, but you can undock it so that it floats or you can dock it to the right or left panel. Be aware that if you dock it to the right or left panel and collapse the panel, you may miss settings when they change. Not that many viewing commands have settings, though, so that may not be a concern.

Thanks,
Nancy McCall

**Reply by ovaisa** [reply]
Is there any documentation/sample on this Tool Settings, how I can use this if I want to create my own toolbars.   Ideally I would prefer to not have this in my UI but if there is no way to remove it, perhaps I can use if for other things too.

Thanks,

Ovais

**Reply by tonygem2022** [reply]
Hi @NancyMcCallB.
Similar query. We wish to disable the Tool Settings tool bar from our UI. 
This tool bar has no benefit to our use cases and only serves to take up screen real estate or confuse because it offers no functionality for what we require. Please advise how to disable it soonest.
Many thanks.
Tony
![Tool Settings](https://github.com/iTwin/itwinjs-core/assets/108276006/b3428063-7a52-4254-bfdf-b5f7bb3e5252)

**Reply by aruniverse** [reply]
Add the following prop to your Viewer component

```tsx
<Viewer
	...
    defaultUiConfig={{
      hideToolSettings: true,
    }}
/>
```

**Comment by ovaisa** [comment]
Hi Arun, thank you, the Toolsetting is no longer loading after I used your suggestion!
